Better integrate with autotools and solve compile warnings

hosiet opened the pull request:
The following list describes my work till now:

* Add a `.gitignore` file to hide annoying autoconf temp files appeared after compilation.
*  Eliminate all compile warnings. This is done by replacing outdated grammar, specifing C language standard (c11) and using feature_test_macros(7).
* Integrate with autotools. Missing checks in configure.ac are added together with various grammar fix.

Please review these changes. Thanks.

sarfata commented:
Thanks a lot. Looks great. 

Just tried this on a Raspbian 7 from a few months back and I get:
````
pi@raspberrypi ~/pi-blaster $ make
make  all-am
make[1]: Entering directory `/home/pi/pi-blaster'
gcc -DHAVE_CONFIG_H -I.     -Wall -pedantic -std=iso9899:2011 -MT pi-blaster.o -MD -MP -MF .deps/pi-blaster.Tpo -c -o pi-blaster.o pi-blaster.c
cc1: error: unrecognized command line option ‘-std=iso9899:2011’
make[1]: *** [pi-blaster.o] Error 1
make[1]: Leaving directory `/home/pi/pi-blaster'
make: *** [all] Error 2
````

What version of GCC do you have?

I have:
````
pi@raspberrypi ~/pi-blaster $ gcc --version
gcc (Debian 4.6.3-14+rpi1) 4.6.3
Copyright (C) 2011 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
````

Also, can you add your name to the list of contributors in the README?

sarfata commented on mailbox.c:
```diff
@@ -40,9 +46,9 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#define dprintf(...) printf(__VA_ARGS__)
 #else
-#define dprintf(args...)
+#define dprintf(...) ;
```
I think you do not need that `;`

sarfata replied:
Why did you keep this `'` here? Is it really needed?

sarfata commented on pi-blaster.c:
```diff
@@ -18,6 +18,16 @@
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#else
+static char VERSION[] = "0.1.1";
```
We are probably going to forget to update this. Maybe we should just replace it with:

````
static char VERSION[] = "SNAPSHOT";
````

hosiet commented:
Sorry for these bugs. I have fixed them in recent commits.

Indeed C11 support was added after gcc 4.7 but Raspbian is using 4.6. For compatibility with Raspbian 7 I switched to `-std=c99`. I myself is using Raspberry Pi B+ with Raspbian upgraded to Debian 8:

```
    pi@raspberrypi ~/src/pi-blaster $ gcc --version
    gcc (Raspbian 4.9.2-10) 4.9.2
    Copyright (C) 2014 Free Software Foundation, Inc.
    This is free software; see the source for copying conditions.  There is NO
    warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

sarfata commented:
:+1: 

Tested it. Works great. Thanks a lot for a great contribution! Please kill the extra `;` (line 51 in `pi-blaster.c`) and let's merge this.

hosiet commented:
Sure. The extra ';' was removed.
